### PR TITLE
Fix scoped search query when file path contains spaces

### LINKED ISSUE #12296: Wrong file: scope when file path contains spaces or special characters
When navigating too a tree/blob, the search field will be scoped to the current repo/file:

image

However, when the file path contains spaces or other special characters, the scoped query is build from the URL-encoded file path, which will return no search results:

image

This is because the file path we use to build the scoped query is [parsed from the location](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@c8c34394d09aa73ac47465d250167f386004c1a4/-/blob/web/src/repo/RepoContainer.tsx?subtree=true#L228-232). 

However, simply fixing `parseBrowserRepoURL` to `decodeURIComponent(filePath)` is not enough: if the decoded file path contains spaces, we would need to quote it to insert it as a valid filter.

### PR DESCRIPTION
**Pull request by lguychard**
Fixes #12296

### CODE REVIEW

File: client/web/src/repo/RepoContainer.tsx [hunk 1]
```diff
@@ -309,7 +310,7 @@ export const RepoContainer: React.FunctionComponent<RepoContainerProps> = props
         } else {
             let query = searchQueryForRepoRevision(repoName, globbing, revision)
             if (filePath) {
-                query = `${query.trimEnd()} file:${globbing ? filePath : '^' + escapeRegExp(filePath)}`
+                query = `${query.trimEnd()} file:${quoteIfNeeded(globbing ? filePath : '^' + escapeRegExp(filePath))}`
```
Comment (felixfbecker) [inline client/web/src/repo/RepoContainer.tsx:313]: @lguychard `quoteIfNeeded()` uses `JSON.stringify()`, which does more than just quoting. Is that not a concern here? I thought there was some issue with it when I tried it (mentioned in the issue) but I can't remember what

Comment (lguychard) [reply client/web/src/repo/RepoContainer.tsx:313]: No specific concern - `quoteIfNeeded()` in its current implementation has been used to quote filter values for a long time. [`searchQueryForRepoRevision()`](), used to build the base `query` here, uses it under the hood. See also [`<RepositoryContributorNode/>`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4ace73fa18f7a7dcca4048b61a67d454e8b5f25a/-/blob/client/web/src/repo/stats/RepositoryStatsContributorsPage.tsx#L46-54), and interactive mode for more examples.

Comment (lguychard) [reply client/web/src/repo/RepoContainer.tsx:313]: Maybe fixing the integration test was what tripped you up? With the additional escaping needed for backslashes in the quoted `file:` filter vs. the unquoted `repo:` filter, and the Monaco-rendered non breaking spaces, it was a bit confusing initially.

Comment (felixfbecker) [reply client/web/src/repo/RepoContainer.tsx:313]: Potentially.
Does it cause a problem with regex mode? If the file includes a dot (file extension), that gets replaced to `\.` by `escapeRegExp()`. But now `JSON.stringify()` sees that and changes it to `\\.`. Is that correct?

Comment (lguychard) [reply client/web/src/repo/RepoContainer.tsx:313]: Yes, it is correct. Consider the following three queries:
- [`repo:^github\.com/sourcegraph/sourcegraph$ f:"hover." type:file`](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+f:%22hover.%22+type:file&patternType=literal): unescaped `.` gets interpreted as a regex metacharacter, results like `hoverAlerts` are shown.
- [`repo:^github\.com/sourcegraph/sourcegraph$ f:"hover\." type:file"`](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+f:%22hover%5C.%22+type:file&patternType=literal): "Unable to process query, unrecognized escape sequence"
- [`repo:^github\.com/sourcegraph/sourcegraph$ f:"hover\\." type:file`](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+f:%22hover%5C%5C.%22+type:file&patternType=literal): shows expected results.

File: client/web/src/integration/repository.test.ts [hunk 2]
```diff
@@ -471,15 +471,17 @@ describe('Repository', () => {
...
-            // await driver.page.waitForSelector('#monaco-query-input .view-lines')
-            // const searchQuery = await driver.page.evaluate(
-            //     () => document.querySelector('#monaco-query-input .view-lines')?.textContent
-            // )
-            // assert.strictEqual(
-            //     searchQuery,
-            //     'repo:^github\\.com/ggilmore/q-test$ file:"^Geoffrey\'s random queries\\.32r242442bf/% token\\.4288249258\\.sql$"'
-            // )
+            await driver.page.waitForSelector('#monaco-query-input .view-lines')
+            // TODO: find a more reliable way to get the current search query,
```
Comment (limitedmage) [inline client/web/src/integration/repository.test.ts:475]: Can we file an issue for this and reference it here?

**Review by limitedmage — APPROVED** [review]

**Review by rvantonder — APPROVED** [review]
LGTM

File: client/web/src/integration/repository.test.ts (hunk 2 above)
Comment (lguychard) [reply client/web/src/integration/repository.test.ts:475]: Yup, done in https://github.com/sourcegraph/sourcegraph/pull/14725/commits/2ae02563c99b246b4bd3dfbc53039e2dfd701b35. Thanks for reminding me of doing this 🙂

**Pull request merged**